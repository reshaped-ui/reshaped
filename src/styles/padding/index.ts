import { responsiveVariables } from "utilities/helpers";
import * as T from "styles/types";
import "./padding.css";

const getPaddingStyles: T.VariableStyleUtility<number> = (value) => {
	if (!value) return null;
	const variables = responsiveVariables("--rs-p", value);

	return { variables };
};

export default getPaddingStyles;
