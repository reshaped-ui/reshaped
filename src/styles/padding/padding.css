@responsive [style*="--rs-p-s:"] {
	@variable --rs-p 0;
}

[style*="--rs-p-s:"] {
	padding: calc(var(--rs-p) * var(--rs-unit-x1));
}
