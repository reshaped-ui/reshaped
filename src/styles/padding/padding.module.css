.root {
	padding: calc(var(--rs-p) * var(--rs-unit-x1) - var(--rs-border-width, 0px));
}

@responsive .root {
	@variable --rs-p 0;
}
