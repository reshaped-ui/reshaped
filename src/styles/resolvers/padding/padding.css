@responsive [style*="--rs-p-s"] {
	@variable --rs-p 0;

	padding: calc(var(--rs-p) * var(--rs-unit-x1));
}

@responsive [style*="--rs-p-top-"] {
	@variable --rs-p-top 0;

	padding-top: calc(var(--rs-p-top) * var(--rs-unit-x1));
}

@responsive [style*="--rs-p-bottom-"] {
	@variable --rs-p-bottom 0;

	padding-bottom: calc(var(--rs-p-bottom) * var(--rs-unit-x1));
}

@responsive [style*="--rs-p-start-"] {
	@variable --rs-p-start 0;

	padding-inline-start: calc(var(--rs-p-start) * var(--rs-unit-x1));
}

@responsive [style*="--rs-p-end-"] {
	@variable --rs-p-end 0;

	padding-inline-end: calc(var(--rs-p-end) * var(--rs-unit-x1));
}

@responsive [style*="--rs-p-block-"] {
	@variable --rs-p-block 0;

	padding-block: calc(var(--rs-p-block) * var(--rs-unit-x1));
}

@responsive [style*="--rs-p-inline-"] {
	@variable --rs-p-inline 0;

	padding-inline: calc(var(--rs-p-inline) * var(--rs-unit-x1));
}
