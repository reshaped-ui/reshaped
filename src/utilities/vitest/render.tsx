import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
