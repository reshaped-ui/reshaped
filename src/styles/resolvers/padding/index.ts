import { responsiveVariables } from "utilities/props";
import * as T from "styles/types";
import "./padding.css";

const padding: T.StyleResolver<T.Padding> = (value) => {
	if (!value) return {};
	const variables = responsiveVariables("--rs-p", value);

	return { variables };
};

export default padding;
