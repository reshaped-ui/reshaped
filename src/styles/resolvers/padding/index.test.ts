import { expect, test, describe } from "vitest";
import padding, {
	paddingBlock,
	paddingBottom,
	paddingEnd,
	paddingInline,
	paddingStart,
	paddingTop,
} from "./index";

describe("Styles/Padding", () => {
	test("handles positive value", () => {
		expect(padding(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(padding(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(padding()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(padding({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Padding/Top", () => {
	test("handles positive value", () => {
		expect(paddingTop(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(paddingTop(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(paddingTop()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(paddingTop({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Padding/Bottom", () => {
	test("handles positive value", () => {
		expect(paddingBottom(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(paddingBottom(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(paddingBottom()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(paddingBottom({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Padding/Start", () => {
	test("handles positive value", () => {
		expect(paddingStart(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(paddingStart(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(paddingStart()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(paddingStart({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Padding/End", () => {
	test("handles positive value", () => {
		expect(paddingEnd(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(paddingEnd(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(paddingEnd()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(paddingEnd({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Padding/Inline", () => {
	test("handles positive value", () => {
		expect(paddingInline(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(paddingInline(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(paddingInline()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(paddingInline({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Padding/Block", () => {
	test("handles positive value", () => {
		expect(paddingBlock(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(paddingBlock(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(paddingBlock()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(paddingBlock({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});
