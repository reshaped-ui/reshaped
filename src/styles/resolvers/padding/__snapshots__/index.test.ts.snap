// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/Padding > handles 0 value 1`] = `{}`;

exports[`Styles/Padding > handles positive value 1`] = `
{
  "variables": {
    "--rs-p-s": 4,
  },
}
`;

exports[`Styles/Padding > handles responsive value 1`] = `
{
  "variables": {
    "--rs-p-l": 2,
    "--rs-p-m": 0,
    "--rs-p-s": 4,
  },
}
`;

exports[`Styles/Padding > handles undefined value 1`] = `{}`;

exports[`Styles/Padding/Block > handles 0 value 1`] = `
{
  "variables": {
    "--rs-p-block-s": 0,
  },
}
`;

exports[`Styles/Padding/Block > handles positive value 1`] = `
{
  "variables": {
    "--rs-p-block-s": 4,
  },
}
`;

exports[`Styles/Padding/Block > handles responsive value 1`] = `
{
  "variables": {
    "--rs-p-block-l": 2,
    "--rs-p-block-m": 0,
    "--rs-p-block-s": 4,
  },
}
`;

exports[`Styles/Padding/Block > handles undefined value 1`] = `{}`;

exports[`Styles/Padding/Bottom > handles 0 value 1`] = `
{
  "variables": {
    "--rs-p-bottom-s": 0,
  },
}
`;

exports[`Styles/Padding/Bottom > handles positive value 1`] = `
{
  "variables": {
    "--rs-p-bottom-s": 4,
  },
}
`;

exports[`Styles/Padding/Bottom > handles responsive value 1`] = `
{
  "variables": {
    "--rs-p-bottom-l": 2,
    "--rs-p-bottom-m": 0,
    "--rs-p-bottom-s": 4,
  },
}
`;

exports[`Styles/Padding/Bottom > handles undefined value 1`] = `{}`;

exports[`Styles/Padding/End > handles 0 value 1`] = `
{
  "variables": {
    "--rs-p-end-s": 0,
  },
}
`;

exports[`Styles/Padding/End > handles positive value 1`] = `
{
  "variables": {
    "--rs-p-end-s": 4,
  },
}
`;

exports[`Styles/Padding/End > handles responsive value 1`] = `
{
  "variables": {
    "--rs-p-end-l": 2,
    "--rs-p-end-m": 0,
    "--rs-p-end-s": 4,
  },
}
`;

exports[`Styles/Padding/End > handles undefined value 1`] = `{}`;

exports[`Styles/Padding/Inline > handles 0 value 1`] = `
{
  "variables": {
    "--rs-p-inline-s": 0,
  },
}
`;

exports[`Styles/Padding/Inline > handles positive value 1`] = `
{
  "variables": {
    "--rs-p-inline-s": 4,
  },
}
`;

exports[`Styles/Padding/Inline > handles responsive value 1`] = `
{
  "variables": {
    "--rs-p-inline-l": 2,
    "--rs-p-inline-m": 0,
    "--rs-p-inline-s": 4,
  },
}
`;

exports[`Styles/Padding/Inline > handles undefined value 1`] = `{}`;

exports[`Styles/Padding/Start > handles 0 value 1`] = `
{
  "variables": {
    "--rs-p-start-s": 0,
  },
}
`;

exports[`Styles/Padding/Start > handles positive value 1`] = `
{
  "variables": {
    "--rs-p-start-s": 4,
  },
}
`;

exports[`Styles/Padding/Start > handles responsive value 1`] = `
{
  "variables": {
    "--rs-p-start-l": 2,
    "--rs-p-start-m": 0,
    "--rs-p-start-s": 4,
  },
}
`;

exports[`Styles/Padding/Start > handles undefined value 1`] = `{}`;

exports[`Styles/Padding/Top > handles 0 value 1`] = `
{
  "variables": {
    "--rs-p-top-s": 0,
  },
}
`;

exports[`Styles/Padding/Top > handles positive value 1`] = `
{
  "variables": {
    "--rs-p-top-s": 4,
  },
}
`;

exports[`Styles/Padding/Top > handles responsive value 1`] = `
{
  "variables": {
    "--rs-p-top-l": 2,
    "--rs-p-top-m": 0,
    "--rs-p-top-s": 4,
  },
}
`;

exports[`Styles/Padding/Top > handles undefined value 1`] = `{}`;
